query-engine/core/src/query_graph_builder/write/nested/upsert_nested.rs

```diff
@@ -169,7 +179,7 @@ pub fn nested_upsert(
                             ))),
                         }?;
 
-                        wq.add_filter(child_id.filter());
+                        ur.set_selectors(vec![child_id]);
```

Comment:
this is minor but it might be better to do `Vec::from([child_id])` 1 less macro call.

Reply:
If you don't mind I kept it this way. We have thousands of `vec!` use, and while I understand the will to reduce compilation times, I think we should either CTRL + R all `vec!` and see if it makes any difference or continue this way meanwhile.

Reply:
Sure makes sense